src/QsCompiler/QirGeneration/DebugInfoManager.cs

```diff
@@ -5,22 +5,38 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.IO;
+using System.Linq;
 using System.Reflection;
 using Microsoft.Quantum.QIR;
+using Microsoft.Quantum.QIR.Emission;
 using Microsoft.Quantum.QsCompiler.DataTypes;
+using Microsoft.Quantum.QsCompiler.SyntaxTokens;
 using Microsoft.Quantum.QsCompiler.SyntaxTree;
 using Ubiquity.NET.Llvm;
 using Ubiquity.NET.Llvm.DebugInfo;
+using Ubiquity.NET.Llvm.Instructions;
+using Ubiquity.NET.Llvm.Types;
+using Ubiquity.NET.Llvm.Values;
 
 namespace Microsoft.Quantum.QsCompiler.QIR
 {
+    using QsTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+    using ResolvedTypeKind = QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>;
+
     /// <summary>
     /// This class holds shared utility routines for generating QIR debug information.
     /// It contains a reference to the GenerationContext that owns it because it needs
     /// access to the same shared state represented by the GenerationContext.
     /// </summary>
     internal sealed class DebugInfoManager
     {
+// SECTION: Config values
```

Comment:
For these configuration options, it might make sense to encapsulate them in a private nested class called `Configuration` or `Options`.

Reply:
If I have a private nested class, then I don't think I can access the member variables in side the class directly (I can't make them public since the class is private) without also creating getters for each config value. This seems like it would make it a bit messy (similar to the getters I had for the CodeviewVersion etc that were really just returning a private string. What do you think?

Reply:
I believe you would be able to do it if the class member variables are public. The class wouldn't be accessible from outside the `DebugInfoManager` class but it would be accessible from inside which is the intent.

I don't think this is blocking so we can go ahead without making this change.